xc/device.ml

```diff
@@ -1102,7 +1102,7 @@ module PCI = struct
     release_xl pcidevs domid
 
   let write_string_to_file file s =
-    let fn_write_string fd = Unixext.really_write fd s 0 (String.length s) in
+    let fn_write_string fd = Unixext.really_write fd (Bytes.unsafe_of_string s) 0 (String.length s) in
```

Comment:
Same here: `s` could be shared.

Reply:
This one I had to check. As above is safe because it is temporarily borrowed to be used by a function that does not mutate it or share it. `really_write` calls `single_write_string` that works like `write`.

On the downside, if somebody replaces the function with something that does share the string, we could have problems. As it is, this relies on the implementation.

Reply:
This function is only used on short strings, it does no harm to make an additional copy. I'll be more conservative and modify this to use `Bytes.of_string`